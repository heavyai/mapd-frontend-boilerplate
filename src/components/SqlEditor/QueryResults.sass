.container
  padding: 0 18px 16px

.table
  border-spacing: 0

  th
    text-align: initial
    padding-bottom: 12px

  td
    padding: 0 3px 3px 0

.error
  color: red

.noResults
  color: #ccc
