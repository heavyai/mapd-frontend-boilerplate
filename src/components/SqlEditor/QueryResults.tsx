import React from 'react'

const styles = require( './QueryResults.sass' )


import { QueryRequest } from '~/reducers/query'

type Props = {
  request: QueryRequest
}


const table = ( columns: string[], data: Object[] ) =>
  <table className={ styles.table }>
    <thead>
      <tr>
        { columns.map( ( column ) => <th key={ column }>{ column }</th> ) }
      </tr>
    </thead>
    <tbody>
      {
        data.map( ( row, index ) =>
          <tr key={ index }>
            {
              columns.map( ( column ) =>
                <td key={ `${ column }-${ index }` }>{ row[ column ] }</td>
              )
            }
          </tr>
        )
      }
    </tbody>
  </table>


const QueryResults: React.StatelessComponent<Props> = ( { request: { results, error } } ) =>
  <div className={ styles.container }>
    { error
      ? <div className={ styles.error }>{ error.message }</div>
      : results && results.length
        ? table( Object.keys( results[ 0 ] ), results )
        : <div className={ styles.noResults }>No results</div>
    }
  </div>


export default QueryResults
