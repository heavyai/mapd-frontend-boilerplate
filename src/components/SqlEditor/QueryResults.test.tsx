import React from 'react'
import { render } from 'enzyme'

import QueryResults from './QueryResults'


test( 'Component renders results', () => {
  const component = render(
    <QueryResults request={ { submitted: true, pending: false, result: [ { col: 'val' } ] } } />
  )

  expect( component ).toMatchSnapshot()
} )

test( 'Component renders error', () => {
  const component = render(
    <QueryResults request={ { submitted: true, pending: false, error: new Error( 'bad' ) } } />
  )

  expect( component ).toMatchSnapshot()
} )
