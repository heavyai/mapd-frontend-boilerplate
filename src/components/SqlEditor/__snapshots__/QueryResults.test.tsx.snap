// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders error 1`] = `
<div
  class="container"
>
  <div
    class="error"
  >
    bad
  </div>
</div>
`;

exports[`Component renders results 1`] = `
<div
  class="container"
>
  <table
    class="table"
  >
    <thead>
      <tr>
        <th>
          col
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          val
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
